UnitySDK/Assets/ML-Agents/Scripts/Agent.cs

```diff
@@ -1011,6 +1021,8 @@ void ResetIfDone()
                             // as it is done
                             _AgentReset();
                             hasAlreadyReset = true;
+                            requestDecision = false;
```

Comment:
How can we make sure this does not break stuff ?

Reply:
Do any of our environments (aside from Snoopy Pop, where this does work) use On Demand Decisions?

Reply:
Bouncer uses ODD

Reply:
Training Bouncer, seems to work (1.0 reward after 50k steps and climbing)